RevenueCatUI/Templates/Components/TemplateComponentsView.swift

```diff
@@ -0,0 +1,202 @@
+//
+//  File.swift
+//  
+//
+//  Created by Josh Holtz on 6/11/24.
+//
+
+import RevenueCat
+import SwiftUI
+// swiftlint:disable all
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+private class ComponentPaywallData: ObservableObject {
+    @Published var selectedPackage: TemplateViewConfiguration.Package
+
+    init(selectedPackage: TemplateViewConfiguration.Package) {
+        self._selectedPackage = .init(initialValue: selectedPackage)
+    }
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct TemplateComponentsView: TemplateViewType {
+
+
+    @StateObject
+    private var componentPaywallData: ComponentPaywallData
+
+    let configuration: TemplateViewConfiguration
+
+    @Environment(\.userInterfaceIdiom)
+    var userInterfaceIdiom
+    @Environment(\.locale)
+    var locale
+
+    #if swift(>=5.9) || (!os(macOS) && !os(watchOS) && !os(tvOS))
+    @Environment(\.verticalSizeClass)
+    var verticalSizeClass
+    #endif
+
+    init(_ configuration: TemplateViewConfiguration) {
+        self.configuration = configuration
+        self._componentPaywallData = .init(wrappedValue: .init(selectedPackage: configuration.packages.default))
```

Comment:
How do people feel about `.init(` as a pattern? I've never really loved it but it might just be that I'm old and used to other languages

Reply:
I don't love it either. I've taken to using it because it seemed to be a standard practice in the codebase, but would be happy to move back to using the name of the item being constructed.